app/lib/bk/compat/parsers/bitrise.rb

```diff
@@ -41,12 +41,29 @@ def parse
 
         stack_value = @config.dig('meta', 'bitrise.io', 'stack')
         agents = stack_value ? { stack: stack_value } : {}
+
+        env = process_app_envs
+
         Pipeline.new(
           steps: bk_steps,
-          agents: agents
+          agents: agents,
+          env: env
         )
       end
 
+      def process_app_envs
+        env = {}
+        (@config.dig('app', 'envs') || []).each do |env_var|
+          env_var.each do |key, value|
+            next if key == 'opts'
+            next if key.to_s.strip.empty? || value.to_s.strip.empty?
+
+            env[key] = value
+          end
+        end
+        env
+      end
+
```

Comment:
It is complicated to follow the logic of this function that just maps the elements removing any `opts` if there are. A simpler version would be:

```suggestion
      def process_app_envs(vars)
        return {} unless vars # takes care of empty vars and nils
        
        envs.map do |env_var|
           env_var.delete('opts')
           
           # if there is more than one thing, it is invalid
           raise unless env_var.one?
           
           env_var.to_a
         end.to_h
      end

```